File: even_fibonacci/php/even_fibonacci.php
<?php

class EvenFibonacci {
	public static function getEvenFibonacciSum($max) {
		$sum = 0;
		$a = 0;
		$b = 1;
		while ($a + $b < $max) {
			if (($a + $b)%2 == 0) {
				$sum += $a + $b;
			}
			$tmp = $a;
			$a = $b;
			$b = $a + $tmp;
		}
		return $sum;
	}
}

var_dump(EvenFibonacci::getEvenFibonacciSum(4000000));

File: even_fibonacci/java/even_fibonacci.java
public class even_fibonacci {
	public static void main(String[] args) {
		int sum=0;
		int a=0,b=1;
		while (a+b<Integer.parseInt(args[0])) {
			if ((a+b)%2 == 0) {
				sum += a+b;
			}
			int tmp=a;
			a=b;
			b=a+tmp;
		}
		System.out.println(sum);
	}
}